Bound install time

cgwalters opened the pull request:
utils: Drop unnecessary lint allow

This function doesn't use unsafe anymore.

Signed-off-by: Colin Walters <walters@verbum.org>

---

boundimage: Use high level `deployment_fd` helper

This keeps things even simpler, it's the same thing we're doing
in `kargs.rs`.

Signed-off-by: Colin Walters <walters@verbum.org>

---

boundimage: More struct definition to top

I think it's generally best to have type definitions come before
the code that references them.

Signed-off-by: Colin Walters <walters@verbum.org>

---

boundimage: Drop const parameter

We were always passing this single constant value to the
function; let's just reference it inside the function
and simplify callers.

Signed-off-by: Colin Walters <walters@verbum.org>

---

boundimage: Rename helper function

I think this name is a bit clearer.

Signed-off-by: Colin Walters <walters@verbum.org>

---

boundimage: Make helpers `pub(crate)`

Prep for using them in the install path.

Signed-off-by: Colin Walters <walters@verbum.org>

---

install: Pull bound images by default

Unless overridden by a CLI option, in which case they will
likely be pulled on firstboot.

Signed-off-by: Colin Walters <walters@verbum.org>

---

cgwalters commented:
OK, this works for me to pull bound images at install time by default.

However:

- I'm not happy with the `/var` situation (xref https://gitlab.com/fedora/bootc/base-images/-/issues/18 too); this also argues somewhat that bound images should be owned in the "bootc storage" as an additional store (also xref https://github.com/containers/bootc/issues/128#issuecomment-2229072527 )
- I think what we're very likely to need in the future is to support a model where we also look for bound images in `/var/lib/containers` in the installation environment, and pull from there if they exist. This would dramatically simplify both local development *and* support hermetic-style builds better
- Needs tests (but so does bound images in general)

ckyrouac reviewed (approved):
lgtm! Validated it works locally via podman-bootc.

cgwalters commented:
OK, rebased :surfer: on git main and lifting draft. I also added a `hide = true` to the install option as bound images are still classified as experimental.
